Tuesdays/Week-Class/Inheratance/Assets/Scripts/Trap.cs

```diff
@@ -30,7 +34,8 @@ public abstract class Trap : MonoBehaviour, IDamagable{
 		//Debug.Log(collision.collider.gameObject.layer);
 		//Debug.Log(whatIDamage & (1 << collision.collider.gameObject.layer));
 		if ((whatIDamage & (1 << collision.collider.GetComponent<PlayerController>().CurrCharacter.gameObject.layer)) != 0){
-			ApplyDamage(collision.collider.gameObject.GetComponent<PlayerController>().CurrCharacter);
+			//ApplyDamage(collision.collider.gameObject.GetComponent<PlayerController>().CurrCharacter);
+			TrapBehaviour(collision.collider.gameObject.GetComponent<PlayerController>(), this);
```

Comment:
you have here a great opportunity for the player to raise an event that he collided with X  
and pass X / X identifier as parameter, and the player. that way player doesn't have to be aware of trap behaviour